### PR TITLE
Server certificate tests

### PR DESCRIPTION
**Pull request by dibell**

### CODE REVIEW

**Comment by dibell** [comment]
@Jc2k review please

File: touchdown/tests/assets/host.crt [hunk 1]
```diff
@@ -0,0 +1,67 @@
+Certificate:
```
Comment (Jc2k) [inline touchdown/tests/assets/host.crt]: I would probably trim this down to the `---BEGIN`

**Comment by Jc2k** [comment]
Not sure what my preferred approach to the unicode issue is. The kind of certs we are dealing are ASCII so can be safely represented as a bytes or as a unicode, so i'd be tempted to just use `touchdown.core.utils.force_bytes` to cast it before passing it the cryptography stuff to validate it. However, if we might support other certificate types (which are binary blobs) then we might have to change the arguments on the resource to be `argument.Bytes` instead of `argument.String`. I'm leaning to switching to `argument.Bytes`.

Looking at botocore, the API deals in strings and not bytes so i'd be tempted to go for the force_bytes option. I think wrapping each `load_pem_x509_certificate(value, backend)` in a `force_bytes()` should do it, and i'd be tempted to drop the `self.private_key.encode('utf-8')` in favour of `force_bytes()` too.

**Comment by Jc2k** [comment]
I'd really like a test that sets a certificate chain. Any rapidssl chain should do the job - as it's invalid it will trigger the error message which has a lot of complicated code to produce a useful error message.

**Pull request merged**